From 2007a2cf5f69412a0ac49bc5a831e616e0f89e1d Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Wed, 29 Jan 2020 01:20:21 +0200
Subject: [PATCH] [SW-8228] goya2: fix RR for disable debug IF

Set all debug IFs to secured in PROT bit and close the ranged registers
for these IFs

Change-Id: If0d61d4fada37a03425d51cb8a45120f68577271
---

diff --git a/common/security.c b/common/security.c
index fff8ff0..738a672 100644
--- a/common/security.c
+++ b/common/security.c
@@ -13,37 +13,46 @@
 
 static void security_disable_i2c_s_mstr_if(void)
 {
-	regmap_writel(0x0,
-		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
-	regmap_writel(0x0,
-		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
-	regmap_writel(0xFFFFFFFF,
-		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-	regmap_writel(0xFF,
-		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-}
-
-static void security_disable_smi_mstr_if(void)
-{
+	/* override AXI prot bit by master IF to secure, unprivlaged access */
+	regmap_writel(0x1,
+		      mmI2C_S_MSTR_IF_CORE_HBW_AXPROT_OVRD);
 	regmap_writel(0x0,
 		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
 	regmap_writel(0x0,
 		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
 	regmap_writel(0xFFFFFFFF,
 		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-	regmap_writel(0xFF,
+	regmap_writel(0x3F,
 		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
+}
+
+static void security_disable_smi_mstr_if(void)
+{
+	/* override AXI prot bit by master IF to secure, unprivlaged access */
+	regmap_writel(0x1,
+		      mmSMI_MSTR_IF_CORE_HBW_AXPROT_OVRD);
+	regmap_writel(0x0,
+		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
+	regmap_writel(0x0,
+		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
+	regmap_writel(0xFFFFFFFF,
+		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
+	regmap_writel(0x3F,
+		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
 }
 
 static void security_disable_jtag_mstr_if(void)
 {
+	/* override AXI prot bit by master IF to secure, unprivlaged access */
+	regmap_writel(0x1,
+		      mmJT_MSTR_IF_CORE_HBW_AXPROT_OVRD);
 	regmap_writel(0x0,
 		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
 	regmap_writel(0x0,
 		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
 	regmap_writel(0xFFFFFFFF,
 		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-	regmap_writel(0xFF,
+	regmap_writel(0x3F,
 		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
 }
 
